bundles/org.openhab.binding.heliosventiremote/src/main/java/org/openhab/binding/heliosventiremote/internal/HeliosVentiRemoteHandler.java

```diff
@@ -0,0 +1,510 @@
+     */
+    public synchronized void polling() {
+        logger.trace("HeliosVentiRemote Polling data for '{}'", getThing().getUID());
+        pollCounter++;
+        if (pollCounter > POLL_OFFLINE_THRESHOLD) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.GONE, "No data received!");
+            disconnect();
+        }
+
+        if (!isConnected()) {
+            connect(); // let's try to reconnect if the connection failed or was never established before
+        }
+
+        variables.values().forEach((v) -> {
+            ChannelUID channelUID = v.channelUID();
+            if (isLinked(channelUID)) {
+                poll(v);
+            }
+        });
+    }
+
+    private void disconnect() {
+        updateStatus(ThingStatus.OFFLINE);
+        if (serialPort != null) {
+            serialPort.close();
+        }
+        IOUtils.closeQuietly(inputStream);
+        IOUtils.closeQuietly(outputStream);
+        outputStream = null;
+        inputStream = null;
+        serialPort = null;
+    }
+
+    private void poll(HeliosVentiRemoteVariable v) {
+        byte txFrame[] = { 0x01, BUSMEMBER_ME, BUSMEMBER_MAINBOARD, 0x00, v.address(), 0x00 };
+        txFrame[5] = (byte) checksum(txFrame);
+
+        tx(txFrame);
+    }
+
+    /*
+     * transmit a frame
+     */
+    private void tx(byte[] txFrame) {
+        try {
+            if (outputStream != null) {
+                logger.trace("HeliosVentiRemote: Write to serial port: {}",
+                        String.format("%02x %02x %02x %02x", txFrame[1], txFrame[2], txFrame[3], txFrame[4]));
+
+                outputStream.write(txFrame);
+                // after each frame we have to wait.
+                // 30 ms is taken from what we roughly see the original remote control is doing
+                Thread.sleep(30);
+            }
+        } catch (IOException e) {
+            // in case we cannot write the connection is somehow broken, let's officially disconnect
+            disconnect();
+            connect();
+        } catch (InterruptedException e) {
+            // ignore if we got interrupted
+        }
+    }
+
+    /**
+     * Check connection status
+     *
+     * @return true if currently connected
+     */
+    private boolean isConnected() {
+        return serialPort != null && inputStream != null && outputStream != null;
+    }
+
+    @Override
+    public void serialEvent(SerialPortEvent event) {
+        switch (event.getEventType()) {
+            case SerialPortEvent.DATA_AVAILABLE:
+                // we get here if data has been received
+                try {
+                    try {
+                        // wait a little to increase the likelihood that the complete frame is already buffered.
+                        // this improves the robustness for RS485/USB converters which sometimes duplicate bytes
+                        // otherwise
+                        Thread.sleep(5);
```

Comment:
We try to avoid sleeping in binding code. I have to think about another way to solve this.

Reply:
Did you find a another way to solve it? - Or how bad is it to wait in the binding?